.NET/Microsoft.Recognizers.Text.DateTime/Japanese/Parsers/JapaneseTimePeriodParserConfiguration.cs

```diff
@@ -75,6 +79,23 @@ public bool GetMatchedTimexRange(string text, out string timex, out int beginHou
             endHour = parseResult.EndHour;
             endMin = parseResult.EndMin;
 
+            // Modify time period if "early"/"late" is present
+            if (DateTimeDefinitions.EarlyHourTermList.Any(o => trimmedText.EndsWith(o, StringComparison.Ordinal)))

```

Comment:
This logic should be the same across all languages. Please move to a common method that receives what the TOD is and returns the range.

Reply:
But this is handled in a different way in standard implementation (looking for 'early'/'late' named groups in the match and modifying the begin/end hours accordingly). 
It has been implemented differently in Japanese because, for example, 'early morning' is not always the combination of 'early' and 'morning' but can be a different word.

Reply:
I meant the mapping between the abstract definition of "morning" or "early morning" into an actual range.
Now this type of code is spread around all languaes. A centralized location should receive the relevant abstracted signals/parameters, not language specific terms, and return a struct/tuple with the range results.
We can start with refactoring CJK, then do the same for the other languages. This will also help in the future to allow users to define their own customized ranges.

Reply:
this seems to be already accomplished by ParseTimeOfDay in TimexUtility (used by both standard and CJK implementation). 
We moved the begin/end hours defined there to Constants

Reply:
I think this is OK in the context of this PR. But there's still a lot of repeated code across culture configs for basically the same thing.